File: IT/Eccezioni/Esercizi/Java/Esercizio7.java
/* file Esercizio7.java */

interface Baz {
	void baz();
}

class Foo implements Baz {
	@Override
	public void baz() {
		System.out.println("Foo");
	}
}

class Bar implements Baz {
	@Override
	public void baz() {
		throw new UnsupportedOperationException("Operazione non supportata!");
	}
}

public class Esercizio7 {
	public static void main(String [] args) {
		Baz [] objs = {new Foo(), null, new Bar(), new Foo()};

		for (Baz obj : objs) {
			try {
				obj.baz();
			}
			catch (NullPointerException e) {
				System.out.println(e.getMessage());
			}
			catch (UnsupportedOperationException e) {
				System.out.println(e.getMessage());
			}
		}

		System.out.println("Fine del programma.");
	}
}

File: IT/Eccezioni/Esercizi/Python/esercizio7.py
# file esercizio7.py

from __future__ import print_function

class Foo(object):
	def baz(self):
		print("Foo")

class Bar(object):
	def baz(self):
		raise NotImplementedError("Operazione non supportata!")

objs = [Foo(), None, Bar(), Foo()]

for obj in objs:
	try:
		obj.baz()
	except AttributeError as e:
		print(e)
	except NotImplementedError as e:
		print(e)

print("Fine del programma.")

